File: code/server/app/models/plan.py
"""
Plan model - 套餐
"""
from sqlalchemy import Column, String, Integer, BigInteger, Numeric, Boolean, DateTime, func
from sqlalchemy.dialects.postgresql import ARRAY
from sqlalchemy.orm import relationship

from app.core.database import Base


class Plan(Base):
    __tablename__ = "plans"
    
    id = Column(String(32), primary_key=True)
    code = Column(String(32), unique=True, nullable=False, index=True)
    name = Column(String(128), nullable=False)
    description = Column(String(512), default="")
    
    # 套餐内容
    traffic_bytes = Column(BigInteger, nullable=False)
    duration_days = Column(Integer, nullable=False)
    price_usd = Column(Numeric(10, 2), nullable=False)
    
    # 支持的支付方式
    supported_assets = Column(ARRAY(String(20)), default=["SOL", "USDT_TRC20"])
    
    # 状态
    enabled = Column(Boolean, default=True, index=True)
    sort_order = Column(Integer, default=0)
    
    # 时间戳
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    # 关系
    orders = relationship("Order", back_populates="plan")
    
    def __repr__(self):
        return f"<Plan {self.code}: {self.name}>"

File: payment_backup/payment/data/model/Plan.kt
package com.v2ray.ang.payment.data.model

import com.google.gson.annotations.SerializedName

/**
 * 套餐数据模型
 */
data class Plan(
    val id: String,
    val name: String,
    val description: String,
    @SerializedName("traffic_bytes")
    val trafficBytes: Long,
    @SerializedName("duration_days")
    val durationDays: Int,
    @SerializedName("price_usd")
    val priceUsd: String,
    @SerializedName("supported_assets")
    val supportedAssets: List<String>,
    val badge: String?
) {
    fun getTrafficDisplay(): String {
        return when {
            trafficBytes >= 1099511627776 -> "${trafficBytes / 1099511627776}TB"
            trafficBytes >= 1073741824 -> "${trafficBytes / 1073741824}GB"
            else -> "${trafficBytes / 1048576}MB"
        }
    }

    fun getDurationDisplay(): String {
        return when {
            durationDays >= 365 -> "${durationDays / 365}年"
            durationDays >= 30 -> "${durationDays / 30}个月"
            else -> "${durationDays}天"
        }
    }

    fun supportsSol(): Boolean = supportedAssets.contains("SOL")
    fun supportsUsdtTrc20(): Boolean = supportedAssets.contains("USDT_TRC20")
}

data class PlansResponse(
    val code: String,
    val message: String,
    val data: PlansData?
)

data class PlansData(
    val plans: List<Plan>
)
